component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java

```diff
@@ -154,6 +155,12 @@ public boolean authenticateClient(HttpServletRequest request, Map<String, List>
                 }
                 registeredCert = (X509Certificate) OAuth2Util
                         .getX509CertOfOAuthApp(oAuthClientAuthnContext.getClientId(), tenantDomain);
+                OAuthAppDO oAuthAppdo = OAuth2Util.getAppInformationByClientId(
+                        oAuthClientAuthnContext.getClientId(), tenantDomain);
+                if (StringUtils.isNotEmpty(oAuthAppdo.getTlsClientAuthSubjectDN()) &&
+                        !requestCert.getSubjectDN().toString().equals(oAuthAppdo.getTlsClientAuthSubjectDN())) {
```

Comment:
Lets invert this to make it null safe

```suggestion
                        !oAuthAppdo.getTlsClientAuthSubjectDN().equals(requestCert.getSubjectDN().toString())) {
```

Reply:
fixed